File: replace_sales_return.js
const fs = require('fs');
const path = "d:\\Billing-app\\billing-app\\src\\app\\dashboard\\sales-return\\create\\page.tsx";

let content = fs.readFileSync(path, "utf-8");

content = content.replace(/CreateSalesInvoice/g, "CreateSalesReturn");
content = content.replace(/Create Sales Invoice/g, "Create Sales Return");
content = content.replace(/Invoice No:/g, "Sales Return No:");
content = content.replace(/Invoice Date:/g, "Sales Return Date:");
content = content.replace(/collection\(db, "invoices"\)/g, 'collection(db, "salesReturns")');
content = content.replace(/type="invoice"/g, 'type="sales-return"');
content = content.replace(/invoiceType === "estimate" \? "Estimate" : "Sales Invoice"/g, '"Sales Return"');
content = content.replace(/value=\{invoiceNumber\}/g, 'value={salesReturnNumber}');
content = content.replace(/setInvoiceNumber/g, 'setSalesReturnNumber');
content = content.replace(/const \[invoiceNumber/g, 'const [salesReturnNumber');
content = content.replace(/value=\{invoiceDate\}/g, 'value={salesReturnDate}');
content = content.replace(/setInvoiceDate/g, 'setSalesReturnDate');
content = content.replace(/const \[invoiceDate/g, 'const [salesReturnDate');
content = content.replace(/salesReturnNumber \? `\$\{salesReturnNumber\}`/g, 'salesReturnNumber ? `${salesReturnNumber}`');

fs.writeFileSync(path, content, "utf-8");
console.log("Done");

File: replace_sales_return.py
import re
import os

path = r"d:\Billing-app\billing-app\src\app\dashboard\sales-return\create\page.tsx"

with open(path, "r", encoding="utf-8") as f:
    content = f.read()

# Replacements
content = content.replace("CreateSalesInvoice", "CreateSalesReturn")
content = content.replace("Create Sales Invoice", "Create Sales Return")
content = content.replace("Invoice No:", "Sales Return No:")
content = content.replace("Invoice Date:", "Sales Return Date:")
content = content.replace('collection(db, "invoices")', 'collection(db, "salesReturns")')
content = content.replace('type="invoice"', 'type="sales-return"')
content = content.replace('invoiceType === "estimate" ? "Estimate" : "Sales Invoice"', '"Sales Return"')
content = content.replace('value={invoiceNumber}', 'value={salesReturnNumber}')
content = content.replace('setInvoiceNumber', 'setSalesReturnNumber')
content = content.replace('const [invoiceNumber', 'const [salesReturnNumber')
content = content.replace('value={invoiceDate}', 'value={salesReturnDate}')
content = content.replace('setInvoiceDate', 'setSalesReturnDate')
content = content.replace('const [invoiceDate', 'const [salesReturnDate')

with open(path, "w", encoding="utf-8") as f:
    f.write(content)
    
print("Replacements done.")
